src/components/molecules/LiveSchedule/LiveSchedule.tsx

```diff
@@ -0,0 +1,86 @@
+import { useSelector } from "hooks/useSelector";
+import { useUser } from "hooks/useUser";
+import React, { FC } from "react";
+import { VenueEvent } from "types/VenueEvent";
+import { isEventLive } from "utils/event";
+import { currentTimeInUnixEpoch, formatHourAndMinute } from "utils/time";
+import { isExternalUrl } from "utils/url";
+import { enterRoom } from "utils/useLocationUpdateEffect";
+import "./LiveSchedule.scss";
+
+const LiveSchedule: FC = () => {
+  const { user, profile } = useUser();
+  const { venueEvents, venue } = useSelector((state) => ({
+    venueEvents: state.firestore.ordered.venueEvents,
+    venue: state.firestore.ordered.currentVenue[0],
+  }));
+
+  const liveEvents =
+    venueEvents && venueEvents.length
+      ? venueEvents.filter((event) => isEventLive(event))
+      : [];
+
+  const onEventClick = (event: VenueEvent) => {
+    const room = venue?.rooms?.find((room) => room.title === event.room);
+
+    const isExternal = isExternalUrl(room.url);
+    if (isExternal) {
+      window.open(room.url, "_blank", "noopener,noreferrer");
+    } else {
+      window.location.href = room.url;
+    }
+    enterRoom(
+      user!,
+      { [`${venue.name}/${room.title}`]: currentTimeInUnixEpoch },
+      profile?.lastSeenIn
+    );
+  };
+
+  return (
+    <div className="schedule-container show">
+      {!!liveEvents.length && (
+        <div className="schedule-day-container">
+          {liveEvents.map((liveEvent, index) => {
+            return (
+              <div
+                key={`live-event-${index}`}
+                className="schedule-event-container schedule-event-container_live"
+              >
+                <div className="schedule-event-time">
+                  <div className="schedule-event-time-start">
+                    {formatHourAndMinute(liveEvent.start_utc_seconds)}
+                  </div>
+                  <div className="schedule-event-time-end">
+                    {formatHourAndMinute(
+                      liveEvent.start_utc_seconds +
+                        60 * liveEvent.duration_minutes
+                    )}
+                  </div>
+                  <span className="schedule-event-time-live">Live</span>
+                </div>
+                <div className="schedule-event-info">
+                  <div className="schedule-event-info-title">
+                    {liveEvent.name}
+                  </div>
+                  <div className="schedule-event-info-description">
+                    {liveEvent.description}
+                  </div>
+                  <div className="schedule-event-info-room">
+                    <div onClick={() => onEventClick(liveEvent)}>
```

Comment:
As above RE: `useCallback`

Reply:
this is still an issue that will cause re-rendering due to the wrapper functions (`() =>`) reference changing on every render

Fix:

Close around the state + `useCallback`:
```typescript
const enterRoom =  useCallback(() => {
    onEventClick(liveEvent)
  }, [liveEvent]);
```

Then use that here:
```suggestion
                    <div onClick={enterRoom}>
```

Reply:
```liveEvent ``` is coming from the iteration of the array ```liveEvents``` and is being passed as parameter to the function. It's not in the scope of the component and I can't simply use it in the useCallback.

Reply:
`liveEvent` needs to be passed as param to the function because it's coming from the iteration of the array.
If the function `enterRoom` is using `useCallback` isn't this going to be the same:
```
onClick={(e) => enterRoom(e)}
onClick={enterRoom}
```

I can probably set the ```liveEvent``` in state and use it from there but then it will cause a re-render because of the setState.

Reply:
Alternatively I can create a small component for the button and have the function there and pass liveEvent as prop, this way it might work.

Reply:
Seems my email response didn't inline here.. but that would be my approach to resolving it too I believe.

fixed in 70e8253242c4c56f3678a5035e0e6566912da044